lib/pf/api.pm

```diff
@@ -222,9 +223,29 @@ sub firewallsso : Public {
 
     my $logger = pf::log::get_logger();
 
-    foreach my $firewall_conf ( sort keys %pf::config::ConfigFirewallSSO ) {
-        my $firewall = pf::factory::firewallsso->new($firewall_conf);
-        $firewall->action($firewall_conf,$postdata{'method'},$postdata{'mac'},$postdata{'ip'},$postdata{'timeout'});
+    foreach my $firewall_id ( sort keys %pf::config::ConfigFirewallSSO ) {
+        my $firewall = pf::factory::firewallsso->new($firewall_id);
+
+        if($postdata{method} eq "Update"){
+            if( pf::util::isenabled($pf::config::ConfigFirewallSSO{$firewall_id}{cache_updates}) ){
+                my $cache = pf::CHI->new( namespace => 'firewall_sso' );
+                my $cache_timeout = $pf::config::ConfigFirewallSSO{$firewall_id}{cache_timeout} || $postdata{timeout} / 2;
```

Comment:
Divide by two is to get close to the renewal.

I want to get the next DHCP packet available to redo the request to handle the expiration on the firewall side.
Basically with this I want it to be the amount of DHCP requests that occur when you don't roam (so one every time the renewal kicks in) 

If I use the lease length then it means I will do it every 2 DHCP packets and multiplying by 2 will mean I will do it every 4.

This is the sweet spot I found in the use case I have so I made it default.
There is still the ability to override if defaults don't apply to the setup

I like the MAC address idea, I will add it